src/lib.rs

```diff
@@ -119,12 +158,27 @@ pub fn reconstruct_from_systematic<'a>(
 /// Works only for 1..65536 chunks.
 /// The data must be non-empty.
 pub fn construct_chunks(n_chunks: u16, data: &[u8]) -> Result<Vec<Vec<u8>>, Error> {
-	if data.is_empty() {
+	if unlikely(data.is_empty()) {
 		return Err(Error::BadPayload);
 	}
-	if n_chunks == 1 {
+	if unlikely(n_chunks == 1) {
 		return Ok(vec![data.to_vec()]);
 	}
+	
+	#[cfg(feature = "arena")]
+	{
+		construct_chunks_arena(n_chunks, data)
+	}
+	
+	#[cfg(not(feature = "arena"))]
+	{
+		construct_chunks_default(n_chunks, data)
+	}
```

Comment:
not a fan of introducing features especially if they are not recommended - what are the expected gains here from the arena allocator?

Reply:
I don’t really understand what kind of starvation you’re talking about. Processing the PoV involves too many different computations to say that it needs all of the CPU time. In the tests, CPU utilization doesn’t even reach 30% of a core for multiple nodes. Asynchronous IO calls, syscalls, etc. let the CPU sit idle, not to mention bubbles when reading from memory.

Even if at peak these computations overlap with some other computations and cause a local slowdown, in the overall workflow the performance gain will be greater than this local collision.